### PR TITLE
Harden the security of Groovy-based DSL, MAL and LAL

### PR DESCRIPTION
**Pull request by kezhenxu94**
- [x] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #<issue number>. NO
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/CHANGES.md).

### CODE REVIEW

**Comment by kezhenxu94** [comment]
This is not all, I just added some usual cases that I could think of, feel free to add more cases

File: oap-server/analyzer/log-analyzer/src/main/java/org/apache/skywalking/oap/log/analyzer/dsl/DSL.java [hunk 1]
```diff
@@ -53,6 +62,22 @@ public static DSL of(final ModuleManager moduleManager,
                 CompileStatic.class
             );
         cc.addCompilationCustomizers(customizer);
+        final SecureASTCustomizer secureASTCustomizer = new SecureASTCustomizer();
+        secureASTCustomizer.setDisallowedStatements(
+            ImmutableList.<Class<? extends Statement>>builder()
+                         .add(WhileStatement.class)
+                         .add(DoWhileStatement.class)
+                         .add(ForStatement.class)
```
Comment (wu-sheng) [inline oap-server/analyzer/log-analyzer/src/main/java/org/apache/skywalking/oap/log/analyzer/dsl/DSL.java:70]: Do we need a `If`?

Comment (kezhenxu94) [reply oap-server/analyzer/log-analyzer/src/main/java/org/apache/skywalking/oap/log/analyzer/dsl/DSL.java:70]: > Do we need a `If`?

We need `if` statement, note this is `DisallowedStatements`

Comment (wu-sheng) [reply oap-server/analyzer/log-analyzer/src/main/java/org/apache/skywalking/oap/log/analyzer/dsl/DSL.java:70]: Oh, I see.

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**